crates/uv/src/commands/pip/list.rs

```diff
@@ -88,21 +87,18 @@ pub(crate) async fn pip_list(
     let latest = if outdated && !results.is_empty() {
         let capabilities = IndexCapabilities::default();
 
-        let client_builder = BaseClientBuilder::new()
-            .retries_from_env()?
-            .connectivity(network_settings.connectivity)
-            .native_tls(network_settings.native_tls)
-            .keyring(keyring_provider)
-            .allow_insecure_host(network_settings.allow_insecure_host.clone());
+        let client_builder = client_builder.clone().keyring(keyring_provider);
```

Comment:
I worry about the friction this adds to adding a keyring provider, i.e., that it makes it easier to forget. I'm not sure what we can do though, it's not substantively worse than the status quo.